pydicom/tests/test_filewriter.py

```diff
@@ -666,22 +666,23 @@ def test_pixel_representation_vm_one(self):
         ref_ds.PixelRepresentation = 0
         ref_ds.SmallestValidPixelValue = b'\x00\x01'  # Little endian 256
         ds = correct_ambiguous_vr(deepcopy(ref_ds), True)
-        self.assertEqual(ds.SmallestValidPixelValue, 256)
-        self.assertEqual(ds[0x00280104].VR, 'US')
+        assert 256 == ds.SmallestValidPixelValue
+        assert 'US' == ds[0x00280104].VR
 
         # If PixelRepresentation is 1 then VR should be SS
         ref_ds.PixelRepresentation = 1
         ref_ds.SmallestValidPixelValue = b'\x00\x01'  # Big endian 1
         ds = correct_ambiguous_vr(deepcopy(ref_ds), False)
-        self.assertEqual(ds.SmallestValidPixelValue, 1)
-        self.assertEqual(ds[0x00280104].VR, 'SS')
+        assert 1 == ds.SmallestValidPixelValue
+        assert 'SS' == ds[0x00280104].VR
 
-        # If no PixelRepresentation then should be unchanged
+        # If no PixelRepresentation AttributeError shall be raised
         ref_ds = Dataset()
         ref_ds.SmallestValidPixelValue = b'\x00\x01'  # Big endian 1
-        ds = correct_ambiguous_vr(deepcopy(ref_ds), True)
-        self.assertEqual(ds.SmallestValidPixelValue, b'\x00\x01')
-        self.assertEqual(ds[0x00280104].VR, 'US or SS')
+        with pytest.raises(AttributeError,
+                           match=r"Failed to resolve ambiguous VR for tag "
+                                 r"\(0028, 0104\):.* 'PixelRepresentation'"):
+            correct_ambiguous_vr(deepcopy(ref_ds), True)
```

Comment:
SmallestValidPixelValue is the element with the ambiguous tag here, not PixelRepresentation